runtime/browser/ui_runtime.cc

```diff
@@ -26,38 +31,86 @@
 #include "common/profiler.h"
 #include "runtime/common/constants.h"
 #include "runtime/browser/native_app_window.h"
+#include "runtime/browser/notification_window.h"
 #include "runtime/browser/preload_manager.h"
-#include "runtime/browser/runtime.h"
 #include "runtime/browser/ui_runtime.h"
 
 namespace runtime {
 
 namespace {
 
-static NativeWindow* CreateNativeWindow() {
+const char kCategoryAlwaysOnTop[] = "http://tizen.org/category/always_on_top";
+
+static NativeWindow* CreateNativeWindow(NativeWindow::Type windowType) {
   SCOPE_PROFILE();
-  // TODO(wy80.choi) : consider other type of native window.
-  auto cached = PreloadManager::GetInstance()->GetCachedNativeWindow();
-  NativeWindow* window = cached != nullptr ? cached : new NativeAppWindow();
+
+  NativeWindow* window;
+
+  switch (windowType) {
+    case NativeWindow::Type::NORMAL: {
+      auto cached = PreloadManager::GetInstance()->GetCachedNativeWindow();
+      window = cached != nullptr ? cached : new NativeAppWindow();
+    }
+    break;
+    case NativeWindow::Type::NOTIFICATION: {
+      PreloadManager::GetInstance()->ReleaseCachedNativeWindow();
+      NotificationWindow* win = new NotificationWindow();
+      window = win;
+    }
+    break;
+  }
+
   window->Initialize();
   return window;
 }
 
+bool HasAlwaysOnTopCategory(
+    const std::vector<std::string>& category_info_list) {
+  for (const auto& category : category_info_list) {
+      if (strcmp(category.c_str(), kCategoryAlwaysOnTop) == 0)
+        return true;
+    }
+    return false;
+}
+
+bool AlwaysTopRequested(common::AppControl* appcontrol) {
+  return appcontrol->data("always_on_top") == "true";
+}
+
 }  // namespace
 
 UiRuntime::UiRuntime(common::ApplicationData* app_data)
-    : application_(NULL),
-      native_window_(NULL),
+    : application_(nullptr),
+      native_window_(nullptr),
+      context_(
+        ewk_context_new_with_injected_bundle_path(INJECTED_BUNDLE_PATH)),
       app_data_(app_data) {
 }
 
 UiRuntime::~UiRuntime() {
-  if (application_) {
-    delete application_;
-  }
-  if (native_window_) {
-    delete native_window_;
-  }
```

Comment:
While app terminating, crash is occurred. 

"application_" should have to be removed before "native_window_" because web_view try to access "native_window _" to remove rotation handler.

reset application_ and native_window_ like below.

application_.reset(nullptr);
native_window_.reset(nullptr);